node/core/pvf/src/execute/worker.rs

```diff
@@ -353,6 +360,11 @@ fn validate_using_artifact(
 	executor: Arc<Executor>,
 	cpu_time_start: ProcessTime,
 ) -> Response {
+	let file_metadata = std::fs::metadata(artifact_path);
+	if let Err(err) = file_metadata {
+		return Response::format_internal("execute: could not find or open file", &err.to_string())
```

Comment:
What if the file is there, but corrupted? Do we handle that case correctly already?

Reply:
Very good question. I believe right now it would result in a panic. There's an open issue about it at https://github.com/paritytech/polkadot/issues/6972

Reply:
Nice! Thank you for staying on top of things! :1st_place_medal: